### DISCUSSION TITLE
Installed extensions are not enabled

### DISCUSSION THREAD

**Discussion by 4r7if3x** [mlocati/docker-php-extension-installer#648] [Q&A] [answered]
I'm using the official PHP docker image with Alpine. As far as I remember, an `ini` file was made per each installed extension by this tool, including `extension=<name>.so` in order to load these modules, but it doesn't happen on my new `PHP 7.4` installation. I'm not quite sure what I'm doing wrong here:

```dockerfile
FROM php:7.4-cli-alpine

ARG BUILD_DEPENDENCIES="freetype-dev gettext-dev icu-dev imagemagick-dev libjpeg-turbo-dev libpng-dev libxml2-dev libxslt-dev libzip-dev oniguruma-dev openssl-dev pcre-dev pcre-dev vips-dev yaml-dev"
ARG PHP_DEPENDENCIES="ca-certificates icu-libs imagemagick libintl libxslt libzip openssl vips yaml"
ARG PHP_EXTENSIONS="bcmath ctype dom exif fileinfo gd intl mbstring mysqli pcntl pdo pdo_mysql sockets sqlite3 zip yaml"
ARG PECL_EXTENSIONS="imagick mcrypt redis"
...
RUN apk add --update --no-cache ${PHP_DEPENDENCIES} \
    && apk add --virtual build-dependencies --no-cache ${PHPIZE_DEPS} ${BUILD_DEPENDENCIES} \
    && docker-php-ext-configure gd --with-freetype --with-jpeg --with-webp \
    && docker-php-ext-install -j$(nproc) ${PHP_EXTENSIONS} \
    && pecl install ${PECL_EXTENSIONS} \
    && docker-php-ext-enable ${PECL_EXTENSIONS} \
    && apk del build-dependencies
...
COPY "./config/php/php.ini" "${PHP_INI_DIR}/php.ini"
```

There are no settings related to extensions on my customized `ini` file, but the files in `/usr/local/etc/php/conf.d/` directory are being loaded, but there is only one file there: `docker-php-ext-sodium.ini`

---

P.S. Do you approve of the way I made a distinction between build & runtime dependencies? + Am I missing something there or with pre-configurations?

**Comment by mlocati** [comment]
This project is about the `install-php-extensions` command, not about the stock `docker-php-ext-configure` / `docker-php-ext-install` / `docker-php-ext-enable` commands.

**Reply by 4r7if3x** [reply]
Oh sorry, I didn't know the difference...

**Comment by mlocati** [comment, marked as answer]
No worries!

PS: what about switching to `install-php-extensions`? You won't need to waste your time determining the dependencies required to build PHP extensions (I already did it for you for [many extensions](https://github.com/mlocati/docker-php-extension-installer#supported-php-extensions)), and, after the extensions have been installed, `install-php-extensions` uninstalls the system packages required only at build time (saving disk space).

**Reply by 4r7if3x** [reply]
Thanks for the clarification. I honestly thought they are the same! :D This actually was one of my own concerns that why dependency management is not included...
